simulator/src/ledger/radix_engine_db.rs

```diff
@@ -0,0 +1,109 @@
+use std::path::{PathBuf};
+
+use rocksdb::{DB, DBWithThreadMode, Direction, IteratorMode, SingleThreaded};
+use radix_engine::ledger::*;
+use radix_engine::model::*;
+use scrypto::types::*;
+
+pub struct RadixEngineDB {
+    db: DBWithThreadMode<SingleThreaded>,
+}
+
+impl RadixEngineDB {
+    pub fn new(root: PathBuf) -> Self {
+        let db = DB::open_default(root.as_path()).unwrap();
```

Comment:
I wonder what happens if the folder already exists but not a database. Most developers will likely have this setup. Need to advise them how to migrate.

Reply:
Default settings creates a new database if empty